FWCore/Framework/test/stubs/ToyIntProducers.cc

```diff
@@ -411,6 +412,39 @@ namespace edmtest {
     e.emplace(otherPutToken_,value);
   }
 
+  //
+  // Produces multiple IntProduct products
+  //
+
+  class ManyIntProducer : public edm::global::EDProducer<> {
+  public:
+    explicit ManyIntProducer(edm::ParameterSet const& p):
+      tokenValues_{vector_transform(p.getParameter<std::vector<edm::ParameterSet>>("values"), [this](edm::ParameterSet const& pset) {
+          return TokenValue{produces<IntProduct>(pset.getParameter<std::string>("instance")),
+                            pset.getParameter<int>("value")};
+        })}
+    {
+      tokenValues_.push_back(TokenValue{produces<IntProduct>(), p.getParameter<int>("ivalue")});
```

Comment:
```suggestion
      tokenValues_.emplace_back(produces<IntProduct>(), p.getParameter<int>("ivalue"));
```

Reply:
That would require the `TokenValue` struct to have a constructor. I can of course do that if you prefer, I went this way to save some repeated typing.

Reply:
Doesn't work unless I add a